improver/metadata/check_datatypes.py

```diff
@@ -38,46 +38,46 @@
     TIME_COORD_NAMES, TIME_INTERVAL_DTYPE, TIME_INTERVAL_UNIT,
     TIME_REFERENCE_DTYPE, TIME_REFERENCE_UNIT)
 
+DESIRED_FLOAT_DTYPE = np.float32
```

Comment:
is this misleading as it'll only be float32 to satisfy the use?

Reply:
Might be a bit overkill yep - just did because having lots of references to `np.float32` seemed a messy (anticipating a future date when `float64` isn't actually a problem...). Happy to remove if required.

(Along the same lines, I should probably have renamed the function `ensure_cube_floats_correct_dtype`, then my case for being more general might have held more water.